### DISCUSSION TITLE
Problem with loss function in Model_0

### DISCUSSION THREAD

**Discussion by kprogia101** [mrdbourke/pytorch-deep-learning#375] [Q&A] [answered]
From the video  learn pytorch in one day, timestamp 6:49:02.
NB: i am in the training phase and i used L1loss as a loss function and i have a problem with the variable "loss".

## My error

RuntimeError: Boolean value of Tensor with more than one value is ambiguous

## My code

```
loss_fn=nn.L1Loss
optimizer=torch.optim.SGD(model.parameters(),lr=0.001)

#reproduce step 
torch.manual_seed(42) 

#training
epochs=1
for epoch in range ( epochs):
  #activer train mode
  model.train()
  #forward
  y_pred=model(train_X)
  #loss
  loss=loss_fn(y_pred,train_Y)
  #metre step à 0
  optimizer.zero_grad()
  #backward
  loss.backward
  #min local<maj poids
  optimizer.step()

``` 

## What I've tried so far
i already checked the stack overflow and they told me i just need to put the loss function in the variable loss_fn which i already did

**Comment by PastorJordi** [comment, marked as answer]
I think you need to add parenthesis `()` after `nn.L1Loss` in order to actually call the method and get the criterion (a.k.a. loss function) returned

**Reply by kprogia101** [reply]
totally worked thank u !

**Comment by mrdbourke** [comment]
Hi @kprogia101,

@PastorJordi is correct. 

Adding `()` to `nn.L1Loss` should help.

For example:

```python
loss_fn=nn.L1Loss()
optimizer=torch.optim.SGD(model.parameters(), lr=0.001)

#reproduce step 
torch.manual_seed(42) 

#training
epochs=1
for epoch in range ( epochs):
  #activer train mode
  model.train()
  #forward
  y_pred=model(train_X)
  #loss
  loss=loss_fn(y_pred,train_Y)
  #metre step à 0
  optimizer.zero_grad()
  #backward
  loss.backward
  #min local<maj poids
  optimizer.step()
```

**Reply by kprogia101** [reply]
thank you so much for your help keep up the great work !

**Reply by IHackmer19** [reply]
I had gotten the same error too.... funny how a `()` can stop the whole code!!
